### PR TITLE
Documentation updates and minor fixes

### PR DESCRIPTION
**Pull request by adrianchiris**
- Update Documentation for project
- Update deployment files for project
- Minor update to Makefile to hide un-needed output
- Fix entrypoint.sh to copy the correct binary to host CNI dir

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,219 @@
...
+that perform [RDMA](https://community.mellanox.com/s/article/what-is-rdma-x) and wish to  leverage network namespace
+isolation of RDMA devices introduced in linux kernel `5.2.0`.
+
+# Requirements
+
+## Hardware
+SR-IOV capable NIC which supports RDMA.
+
+### Supported Hardware
+
```
Comment (moshe010) [inline README.md]: need add support  FW latest GA and above

File: README.md [hunk 2]
```diff
@@ -1,13 +1,219 @@
...
+#### Mellanox Network adapters
+- ConnectX®-4 Lx
```
Comment (moshe010) [inline README.md]: we support ConnectX®-4 and above (The limitation in sriov device plugin is because of redhat it not real limitation)

File: README.md [hunk 3]
```diff
@@ -1,13 +1,219 @@
...
+1. Copy `rdma` binary to the CNI dir in each worker node.
+2. Build container image, push to your own image repo then modify the deployment template and deploy. 
 
-## At high level
-RDMA CNI plugin is intended to be run as a chained CNI plugin (introduced in CNI Specifications `v0.3.0`) moving the associated
-RDMA interfaces of a given network interface to the provided network namespace path
+#### Run tests:
+```console
+~$ make tests
+running gofmt...
```
Comment (moshe010) [inline README.md]: I prefer not to put the console output, because if we change it we probably for get to change it here

File: README.md [hunk 4]
```diff
@@ -1,13 +1,219 @@
...
+```
 
-## Links
-https://github.com/containernetworking/cni/blob/v0.3.0/SPEC.md#network-configuration
+#### Build image:
+```console
+~$ make image
+Building Docker image...
+## Build conatiner image
+Sending build context to Docker daemon  113.4MB
```
Comment (moshe010) [inline README.md]: same as above

File: deployment/rdma-cni-daemonset-pre-1.16.yaml [hunk 5]
```diff
@@ -8,22 +8,22 @@ metadata:
     tier: node
     app: rdma-cni
 spec:
+  updateStrategy:
+    type: RollingUpdate
```
Comment (moshe010) [inline deployment/rdma-cni-daemonset-pre-1.16.yaml]: why this change?
Also I am not sure we need the pre1.16 kubernetes. It don't have topology manager so none will use SR-IOV with max performance

File: deployment/rdma-cni-daemonset.yaml [hunk 6]
```diff
@@ -7,27 +7,28 @@ metadata:
   labels:
     tier: node
     app: rdma-cni
+    name: rdma-cni
 spec:
   selector:
     matchLabels:
       name: rdma-cni
+  updateStrategy:
+    type: RollingUpdate
```
Comment (moshe010) [inline deployment/rdma-cni-daemonset.yaml]: same comment as above

File: deployment/rdma-cni-daemonset.yaml [hunk 7]
```diff
@@ -7,27 +7,28 @@ metadata:
...
     metadata:
       labels:
         tier: node
         app: rdma-cni
         name: rdma-cni
     spec:
-      hostNetwork: true
       nodeSelector:
-        beta.kubernetes.io/arch: amd64
+        kubernetes.io/arch: amd64
```
Comment (moshe010) [inline deployment/rdma-cni-daemonset.yaml]: I don't think this is need if we use annotation

File: images/README.md [hunk 8]
```diff
@@ -5,7 +5,7 @@ This is used for distribution of RDMA CNI binary in a Docker image.
 Typically you'd build this from the root of your RDMA CNI clone, and you'd set the `DOCKERFILE` to specify the Dockerfile during build time, `TAG` to specify the image's tag:
 
 ```
-$ DOCKERFILE=Dockerfile TAG=Mellanox/rdma-cni make image
+$ DOCKERFILE=Dockerfile TAG=mellanox/rdma-cni make image
```
Comment (moshe010) [inline images/README.md]: is rdma or rdma-cni. Sometimes  is rdma and sometime rdma-cni

File: README.md (hunk 1 above)
Comment (adrianchiris) [reply README.md]: why is that needed ? because of what we test with ? rdma namespace aware does not depend on the NIC Firmware to my knowledge.

File: README.md (hunk 2 above)
Comment (adrianchiris) [reply README.md]: will Update to ConnectX-4 Family devices or newer

File: README.md (hunk 3 above)
Comment (adrianchiris) [reply README.md]: will change

File: README.md (hunk 4 above)
Comment (adrianchiris) [reply README.md]: will change

File: deployment/rdma-cni-daemonset-pre-1.16.yaml (hunk 5 above)
Comment (adrianchiris) [reply deployment/rdma-cni-daemonset-pre-1.16.yaml]: Thats how multus does it. see : https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/

the tl;dr; is : with Rolling update you dont need to delete the pods when you update the daemonset

File: deployment/rdma-cni-daemonset.yaml (hunk 7 above)
Comment (adrianchiris) [reply deployment/rdma-cni-daemonset.yaml]: you mean, if we use docker manifest ?

with docker manifest we will need just one definition of daemonset in the deployment, so ill remove the dedicate ppc64le

File: deployment/rdma-cni-daemonset-pre-1.16.yaml (hunk 5 above)
Comment (adrianchiris) [reply deployment/rdma-cni-daemonset-pre-1.16.yaml]: the pre-1.16 can be removed if not needed, but we will need to update the README to state that kubernetes version is expected to be >= 1.16

File: images/README.md (hunk 8 above)
Comment (adrianchiris) [reply images/README.md]: rdma-cni - is the image name
rdma - is the binary name

File: deployment/rdma-cni-daemonset-pre-1.16.yaml (hunk 5 above)
Comment (moshe010) [reply deployment/rdma-cni-daemonset-pre-1.16.yaml]: Ok let do that 1.16 and above

File: README.md (hunk 1 above)
Comment (moshe010) [reply README.md]: Ok dont add it

File: README.md [hunk 9]
```diff
@@ -1,13 +1,219 @@
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0)
+[![Go Report Card](https://goreportcard.com/badge/github.com/Mellanox/rdma-cni)](https://goreportcard.com/report/github.com/Mellanox/rdma-cni)
+[![Build Status](https://travis-ci.com/Mellanox/rdma-cni.svg?branch=master)](https://travis-ci.com/Mellanox/rdma-cni)
+[![Coverage Status](https://coveralls.io/repos/github/Mellanox/rdma-cni/badge.svg)](https://coveralls.io/github/Mellanox/rdma-cni)
+
```
Comment (moshe010) [inline README.md]: I think all of this should be in one line see [1]

[1] - https://raw.githubusercontent.com/intel/multus-cni/master/README.md
